x/upgrade/abci.go

```diff
@@ -12,6 +12,16 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
 )
 
+// downgradeVerified is a variable that tells if we've already sanity checked that this binary version isn't being used against an old state.
+// If so, we dont run that check every beginblock.
+var downgradeVerified = false
```

Comment:
can we have this be part of the keeper instead? Or is that not possible? I just really try to advise against the use of globals when possible.

Reply:
Aren't keepers supposed to be stateless though?

Reply:
Kinda sorta, yes. Staking isn't (hooks). This is just a suggestion though.

Reply:
Putting this variable in keeper is better than using global.
Why keepers should be stateless?